utilities/src/main/java/io/syndesis/qe/resource/impl/Syndesis.java

```diff
@@ -705,4 +776,113 @@ public void linkServiceAccountWithSyndesisPullSecret(String serviceAccountName)
+        }
+        Index index;
+        Bundle foBundle = null;
+        OpenShiftService ocpSvc;
+        Opm opm = new Opm();
+        QuayUser quayUser = TestConfiguration.getQuayUser();
+        if (TestConfiguration.getBundleImage() != null) {
+            //Deploy from bundle
+            index = opm.createIndex("quay.io/marketplace/fuse-online-index:" + TestConfiguration.syndesisVersion());
+            foBundle = index.addBundle(TestConfiguration.getBundleImage());
+            index.push(quayUser);
+            ocpSvc = TestConfiguration.getOpenShiftService("fuse-online-index");
+        } else {
+            //deploy from existing index image
+            index = opm.pullIndex(TestConfiguration.getIndexImage(), quayUser);
+            index.push(quayUser);
+            String[] parts = TestConfiguration.getIndexImage().split("/");
+            String quayProject = parts[parts.length - 1].split(":")[0];
+            ocpSvc = TestConfiguration.getOpenShiftService(quayProject);
+        }
+        try {
+            // OCP stuff - add index
+            ocpSvc.patchGlobalSecrets(TestConfiguration.getQuayPullSecret());
+            index.addIndexToCluster(ocpSvc, "fuse-online-test-catalog");
+            if (foBundle != null) {
+                foBundle.createSubscription(ocpSvc);
+            } else {
+                Bundle.createSubscription(ocpSvc, "fuse-online", "fuse-online-v7.8.x", "''", "fuse-online-test-catalog");
+            }
+        } catch (IOException | TimeoutException | InterruptedException e) {
+            e.printStackTrace();
+        }
+        createPullSecret();
+        if (TestConfiguration.enableTestSupport()) {
+            enableTestSupport();
+        }
+        deployCrAndRoutes();
+        CommonSteps.waitForSyndesis();
+    }
+
+    private void enableTestSupport() {
+        TestUtils.waitFor(() -> "AtLatestKnown".equalsIgnoreCase(getSubscription().getJSONObject("status").getString("state")), 2, 60 * 3,
+            "CSV didn't get installed in time");
+        JSONObject json = new JSONObject(
+            OpenShiftUtils.getInstance().customResource(getCSVContext()).get(TestConfiguration.openShiftNamespace(), "fuse-online-operator.v7.8.0"));
+        JSONObject operatorDeployment =
+            json.getJSONObject("spec").getJSONObject("install").getJSONObject("spec").getJSONArray("deployments").getJSONObject(0);
+        JSONArray envVars =
+            operatorDeployment.getJSONObject("spec").getJSONObject("template").getJSONObject("spec").getJSONArray("containers").getJSONObject(0)
+                .getJSONArray("env");
+        envVars.put(TestUtils.map("name", "TEST_SUPPORT", "value", "true"));
+        try {
+            OpenShiftUtils.getInstance().customResource(getCSVContext())
+                .edit(TestConfiguration.openShiftNamespace(), "fuse-online-operator.v7.8.0", json.toMap());
+        } catch (IOException e) {
+            log.error("Couldn't edit Syndesis CSV", e);
+        }
+    }
```

Comment:
there is a code for that here: https://github.com/syndesisio/syndesis-qe/blob/master/utilities/src/main/java/io/syndesis/qe/TestSuiteParent.java#L37-L60

please merge it into one method that will do the following:
- TestSuiteParent will call that method
- in that method, check what type of deployment it is - either "standard" or operator hub
- for operator hub, check if the test support is already enabled, if no, enable it
- for standard deployment, check `if TestConfiguration.enableTestSupport()` and if yes, do that part of the code

Reply:
how does one merge? :smile: 
I think enabling test support needs to happen at different times for both ways. I need to edit the CSV after creating a subscription to ensure CSV is present on the cluster. And this code looks like it needs to run even prior to the operator existing.

Reply:
you're right, my bad :grin: